content/token-swap.md

```diff
@@ -0,0 +1,848 @@
+12. `tokenProgramId` - the address of the token program
+13. `amountIn` - amount of tokens deposited by the user into the swap pool
+14. `minimumAmountOut` - minimum amount of tokens send to the user token account. This parameter is used to account for slippage. Slippage is the difference between the value of a token when you submit the transaction versus when the order is fulfilled. In this case, the lower the number, the more slippage can possible occur without the transaction failing. Throughout this lesson we'll use 0 for swaps as calculating slippage is outside the scope of this lesson. In a production app, however, it's important to let users specify the amount of slippage they're comfortable with.
+
+The instruction for swapping token A for token B will look like this:
+
+```tsx
+const swapInstruction = TokenSwap.swapInstruction(
+    tokenSwapStateAccount,
+    swapAuthority,
+    userPublicKey,
+    userTokenA,
+    poolTokenA,
+    poolTokenB,
+    userTokenB,
+    poolMint,
+    feeAccount,
+    null,
+    TOKEN_SWAP_PROGRAM_ID,
+    TOKEN_PROGRAM_ID,
+    amount * 10 ** MintInfoTokenA.decimals,
+    0
+)
+
+transaction.add(swapInstruction)
+```
+
+### Deposit liquidity
+
+The token swap program has two variations of deposit instructions. One allows users to only deposit tokens to one side of the swap pool at a time. The other allows for users to deposit to both sides of the swap pool at the same time.
+
+In order to deposit liquidity to both sides of the swap pool, a user’s wallet must have a sufficient amount of each token. When depositing both tokens, instead of providing the amount of each token to deposit, the user specifies the amount of LP-tokens they would like to receive. The token swap program then calculates the amount of each token that a depositor will receive given the pool's Curve and current liquidity.
+
+We can deposit both tokens at the same time using the `TokenSwap.depositAllTokenTypesInstruction` helper function which requires the following arguements:
+1. `tokenSwap` - the token swap state account
+2. `authority` - the swap pool authority
+3. `userTransferAuthority` - the authority over the user token accounts
+4. `sourceA` - user token A account to transfer tokens into the swap pool token A account
+5. `sourceB` - user token B account to transfer tokens into the swap pool token B account
+6. `intoA` - swap pool token account A to receive user's token A
+7. `intoB` - swap pool token account B to receive user's token B
+8. `poolToken` - the LP-token mint address
+9. `poolAccount` - user LP-token account the swap pool mints LP-token to
+10. `swapProgramId` - the address of the token swap program
+11. `tokenProgramId` - the address of the token program
+12. `poolTokenAmount` - amount of LP-token the depositor expects to receive
+13. `maximumTokenA` - maximum amount of token A allowed to deposit
+14. `maximumTokenB` - maximum amount of token A allowed to deposit.
```

Comment:
@jamesrp13 the period at the end of this list is acceptable, but it feels awkward to me.

Reply:
Nope not acceptable haha